Sources/Turf/Geometries/LineString.swift

```diff
@@ -215,4 +215,132 @@ extension LineString {
+            }
+        }
+    }
+
+    private func simplifyDouglasPeucker(_ coordinates: [CLLocationCoordinate2D], tolerance: Double) -> [CLLocationCoordinate2D] {
+        if coordinates.count <= 2 {
+            return coordinates
+        }
+
+        let lastPoint = coordinates.count - 1
+        var result = [coordinates[0]]
+        simplifyDouglasPeuckerStep(coordinates, first: 0, last: lastPoint, tolerance: tolerance, simplified: &result)
+        result.append(coordinates[lastPoint])
+        return result
+    }
+
+    /// Returns a simplified version of the LineString using the Ramer–Douglas–Peucker algorithm.
+    ///
+    /// tolerance:  Controls the level of simplification by specifying the maximum allowed distance between the original line point
+    /// and the simplified point. Higher tolerance values results in higher simplification.
+    /// 
+    /// highestQuality: Excludes distance-based preprocessing step which leads to highest quality simplification. High quality simplification runs considerably slower so consider how much precision is needed in your application.
+    ///
+    /// Ported from https://github.com/Turfjs/turf/blob/master/packages/turf-simplify/lib/simplify.js
+    public func simplified(tolerance: Double = 1.0, highestQuality: Bool = false) -> LineString {
+        guard coordinates.count > 2 else { return LineString(coordinates) }
+
+        let squareTolerance = tolerance * tolerance
+
+        var simplifiedCoordinates = highestQuality ? coordinates : self.simplified(radialTolerance: squareTolerance).coordinates
+        
+        simplifiedCoordinates = simplifyDouglasPeucker(simplifiedCoordinates, tolerance: squareTolerance)
+
+        return LineString(simplifiedCoordinates)
+    }
+
+    /// Mutates the LineString into a simplified version using the Ramer–Douglas–Peucker algorithm.
+    ///
+    /// tolerance:  Controls the level of simplification by specifying the maximum allowed distance between the original line point
+    /// and the simplified point. Higher tolerance values results in higher simplification.
+    ///
+    /// highestQuality: Excludes distance-based preprocessing step which leads to highest quality simplification. High quality simplification runs considerably slower so consider how much precision is needed in your application.
+    ///
+    /// Ported from https://github.com/Turfjs/turf/blob/master/packages/turf-simplify/lib/simplify.js
+    public mutating func simplified(tolerance: Double = 1.0, highestQuality: Bool = false) {
+        guard coordinates.count > 2 else { return }
+
+        let squareTolerance = tolerance * tolerance
+
+        let simplifiedCoordinates = highestQuality ? coordinates : self.simplified(radialTolerance: squareTolerance).coordinates
+
+        coordinates = simplifyDouglasPeucker(simplifiedCoordinates, tolerance: squareTolerance)
```

Comment:
This still ends up copying the LineString before mutating the original, because `simplified(radialTolerance:)` creates a new LineString. Ideally, there should be a mutating method `simplified(tolerance:highestQuality:)` that operates directly on the current coordinate array, plus a non-mutating method `simplify(tolerance:highestQuality:)` method that copies self and calls `simplified(tolerance:highestQuality:)` on it.

Reply:
Should be fixed. I also did add mutating and non-mutating variants.